pkg/signals/signal.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2017 The Kubernetes Authors.
```

Comment:
Why not just import this?

Reply:
This is a slightly modified version from the original source ([here](https://github.com/kubernetes/sample-controller/blob/master/pkg/signals/signal.go)).  This version adds support for calling the `CancelFunc`, from a context, as well as the signals

Reply:
I don't think that's how the copyright works. You need to have a `third_party` for the original code and carry patches IIRC. But honestly if it's good enough for k8s ... probably good enough for you :)